solutions/src/test/java/ru/sberbank/school/task10/ConcurrentScalableThreadPoolTest.java

```diff
@@ -0,0 +1,169 @@
+        latch.await();
+
+        latch = new CountDownLatch(25);
+        for (int i = 0; i < 25; i++) {
+            TimeUnit.MILLISECONDS.sleep(6);
+            threadPool.execute(newRunnable(5, counter, latch));
+        }
+
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+        latch.await();
+
+        assertEquals(50, counter.get());
+
+        threadPool.stopNow();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(nowThreadSize, initialThreadSize);
+    }
+
+    @Test
+    public void callableService() throws InterruptedException, ExecutionException {
+        final AtomicInteger counter = new AtomicInteger(0);
+        ArrayList<Future<String>> futureList = new ArrayList<>();
+        Thread.sleep(10);
+        int initialThreadSize = Thread.getAllStackTraces().keySet().size();
+        CountDownLatch latch;
+        int nowThreadSize;
+
+        threadPool.start();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        latch = new CountDownLatch(25);
+        for (int i = 0; i < 25; i++) {
+            Future<String> future = threadPool.execute(newCallable(100, counter, latch));
+            futureList.add(future);
+        }
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(max, nowThreadSize - initialThreadSize);
+
+        for (Future future : futureList) {
+            System.out.println(future.get());
+        }
+
+        latch.await();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        assertEquals(25, counter.get());
+
+        threadPool.stopNow();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(nowThreadSize, initialThreadSize);
+    }
+
+    @Test
+    public void executeAfterStop() {
+        threadPool.start();
+        threadPool.stopNow();
+        Assertions.assertThrows(IllegalStateException.class, () ->
+                threadPool.execute(() -> {
+                }));
+    }
+
+    @Test
+    public void doubleStop() {
+        threadPool.start();
+        threadPool.stopNow();
+        Assertions.assertThrows(IllegalStateException.class, () -> threadPool.stopNow());
+    }
+
+    @Test
+    public void doubleStart() {
+        threadPool.start();
+        Assertions.assertThrows(IllegalStateException.class, () -> threadPool.start());
+    }
+
+    @Test
+    public void wrongThreadsCount() {
+        Assertions.assertAll(
+                () -> Assertions.assertThrows(IllegalArgumentException.class
+                        , () -> new ConcurrentScalableThreadPool(0, 4)),
```

Comment:
&#39;,&#39; should be on the previous line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1928780834>*